src/wp-includes/class-wp-query.php

```diff
@@ -3454,7 +3454,7 @@ public function get_posts() {
 			/** @var WP_Post[] */
 			$this->posts = array_map( 'get_post', $this->posts );
 
-			if ( $q['cache_results'] ) {
+			if ( $q['cache_results'] && ! wp_using_ext_object_cache() ) {
 				update_post_caches( $this->posts, $post_type, $q['update_post_term_cache'], $q['update_post_meta_cache'] );
 			}
```

Comment:
Does this work as an alternative, it will only cache uncached post IDs.

`$post_ids = wp_list_pluck( $this->posts, 'ID' );` is run above but it would be good to confirm it's defined at this point under all possible branches.

```suggestion
			if ( $q['cache_results'] ) {
				_prime_post_caches( $post_ids, $q['update_post_term_cache'], $q['update_post_meta_cache'] );
			}
```

Reply:
@peterwilsoncc I tested this changes but it break unit tests https://github.com/WordPress/wordpress-develop/pull/3353

Reply:
To honest, I am not even sure of the point of this line of code anyway. It is really confusing. Priming the caches conditionally seems weird when you already have `update_post_term_cache` and `update_post_meta_cache` for this. 

I would be happy removing `if ( $q['cache_results'] ) {` check all together. Thoughts @mukeshpanchal27 @peterwilsoncc ?

Reply:
I think the switch to `_prime_post_caches()` wasn't a good idea, see https://core.trac.wordpress.org/ticket/57163.